libcontainer/cgroups/fs/cpuset.go

```diff
@@ -49,7 +52,112 @@ func (s *CpusetGroup) Remove(d *cgroupData) error {
 	return removePath(d.path("cpuset"))
 }
 
+func getCpusetStat(path string, filename string) ([]uint16, error) {
+	var extracted []uint16
+	fileContent, err := fscommon.GetCgroupParamString(path, filename)
+	if err != nil {
+		return extracted, err
+	}
+	if len(fileContent) == 0 {
+		return extracted, fmt.Errorf("%s found to be empty", filepath.Join(path, filename))
+	}
+
+	for _, s := range strings.Split(fileContent, ",") {
+		if !strings.Contains(s, "-") {
+			value, err := strconv.ParseUint(s, 10, 16)
+			if err != nil {
+				return extracted, err
+			}
+			extracted = append(extracted, uint16(value))
+
+		} else {
+			splitted := strings.Split(s, "-")
+			if len(splitted) != 2 {
+				return extracted, fmt.Errorf("Invalid values in %s", filepath.Join(path, filename))
+			}
+			min, err := strconv.ParseUint(splitted[0], 10, 16)
+			if err != nil {
+				return extracted, err
+			}
+			max, err := strconv.ParseUint(splitted[1], 10, 16)
+			if err != nil {
+				return extracted, err
+			}
+			if min > max {
+				return extracted, fmt.Errorf("Invalid values in %s", filepath.Join(path, filename))
+			}
+			for i := min; i <= max; i++ {
+				extracted = append(extracted, uint16(i))
+			}
+		}
+
+	}
+
+	return extracted, nil
+}
+
 func (s *CpusetGroup) GetStats(path string, stats *cgroups.Stats) error {
+
+	var errStrings []string
+
+	if cpus, err := getCpusetStat(path, "cpuset.cpus"); err != nil {
+		errStrings = append(errStrings, err.Error())
```

Comment:
I think it does not make sense to try all the files and then still return an error. I mean, you can just return the first error you have encountered (unless I am missing something here).

Reply:
@piowag please let me know what do you think about the above suggestion

Reply:
Returning all errors seems better for user trying to debug the issue.